libs/wingsdk/src/utils/resource-names.ts

```diff
@@ -0,0 +1,38 @@
+export enum CaseConventions {
+  LOWERCASE = "lowercase",
+  UPPERCASE = "uppercase",
+}
+
+export interface NameOptions {
+  readonly resource?: string;
+  readonly regexMatch: RegExp;
+  readonly charReplacer: string;
+  readonly maxLen: number;
+  readonly minLen?: number;
+  readonly case?: CaseConventions;
+  readonly avoidStartWith?: string;
+  readonly prefix?: string;
+}
+
+export class ResourceNames {
+  public static of(name: string, props: NameOptions): string {
```

Comment:
As mentioned in a previous review, the names should be constructed such that the `node.addr` is always included because it ensures that there will be no duplicates throughout the app (it is a hash of the path of the construct).

This implies that to `maxLen` should only effect the "human-readable" portion of the name, and not the address.

So a name should usually look like this:

```ts
const hash = node.addr;
const human = node.id.substring(0, props.maxLen - hash.length);
return ${human}-${hash};
```

(check my math!)

I would recommend passing in the construct node instead of `name: string`, so that the formatting logic can be centralized here.

If you go back to CDK8s implementation, you'll see this logic as well.

Reply:
@eladb 
I saw the limits of Bucket name size on Azure and max length was 24 characters, currently hash has 42 characters

If we apply the cut only on id I think it's not be userfull to Azure

Reply:
@marciocadev It should be okay to use a small suffix of the hash (like `node.addr.substring(-8)`) since that should still provide enough randomness to avoid conflicts

Reply:
good, I think that leaves plenty of room for a descriptive name